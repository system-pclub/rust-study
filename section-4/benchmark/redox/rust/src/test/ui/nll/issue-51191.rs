struct Struct;

impl Struct {
    fn bar(self: &mut Self) {
        //~^ WARN function cannot return without recursing
        (&mut self).bar();
        //~^ ERROR cannot borrow `self` as mutable, as it is not declared as mutable [E0596]
    }

    fn imm(self) {
        (&mut self).bar();
        //~^ ERROR cannot borrow `self` as mutable, as it is not declared as mutable [E0596]
    }

    fn mtbl(mut self) {
        (&mut self).bar();
    }

    fn immref(&self) {
        (&mut self).bar();
        //~^ ERROR cannot borrow `self` as mutable, as it is not declared as mutable [E0596]
        //~^^ ERROR cannot borrow data in a `&` reference as mutable [E0596]
    }

    fn mtblref(&mut self) {
        (&mut self).bar();
        //~^ ERROR cannot borrow `self` as mutable, as it is not declared as mutable [E0596]
    }
}

fn main () {}
