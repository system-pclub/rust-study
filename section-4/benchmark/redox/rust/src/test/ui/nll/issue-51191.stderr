warning: function cannot return without recursing
  --> $DIR/issue-51191.rs:4:5
   |
LL |     fn bar(self: &mut Self) {
   |     ^^^^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
LL |
LL |         (&mut self).bar();
   |         ----------------- recursive call site
   |
   = note: `#[warn(unconditional_recursion)]` on by default
   = help: a `loop` may express intention better if this is on purpose

error[E0596]: cannot borrow `self` as mutable, as it is not declared as mutable
  --> $DIR/issue-51191.rs:6:9
   |
LL |         (&mut self).bar();
   |         ^^^^^^^^^^^
   |         |
   |         cannot borrow as mutable
   |         try removing `&mut` here

error[E0596]: cannot borrow `self` as mutable, as it is not declared as mutable
  --> $DIR/issue-51191.rs:11:9
   |
LL |     fn imm(self) {
   |            ---- help: consider changing this to be mutable: `mut self`
LL |         (&mut self).bar();
   |         ^^^^^^^^^^^ cannot borrow as mutable

error[E0596]: cannot borrow `self` as mutable, as it is not declared as mutable
  --> $DIR/issue-51191.rs:20:9
   |
LL |         (&mut self).bar();
   |         ^^^^^^^^^^^ cannot borrow as mutable

error[E0596]: cannot borrow data in a `&` reference as mutable
  --> $DIR/issue-51191.rs:20:9
   |
LL |         (&mut self).bar();
   |         ^^^^^^^^^^^ cannot borrow as mutable

error[E0596]: cannot borrow `self` as mutable, as it is not declared as mutable
  --> $DIR/issue-51191.rs:26:9
   |
LL |         (&mut self).bar();
   |         ^^^^^^^^^^^
   |         |
   |         cannot borrow as mutable
   |         try removing `&mut` here

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0596`.
